### PR TITLE
Remove deprecated constructors of URL in SpashScreen

### PR DESCRIPTION
**Pull request by SWinxy**
URL constructors have been deprecated in Java 20, and are still used in SplashScreen. This change replaces the constructors of URL to be a URI constructor. Splish splash. I couldn't verify that finding an image in the jar works in either Java 19 or in a build of this change. `java -splash:image.png -jar path/to/my.jar` failed to show an image on the two versions I tested.

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [ ] Commit message must refer to an issue

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11891/head:pull/11891` \
`$ git checkout pull/11891`

Update a local copy of the PR: \
`$ git checkout pull/11891` \
`$ git pull https://git.openjdk.org/jdk pull/11891/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11891`

View PR using the GUI difftool: \
`$ git pr show -t 11891`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11891.diff">https://git.openjdk.org/jdk/pull/11891.diff</a>

</details>

### CODE REVIEW

File: src/java.desktop/share/classes/java/awt/SplashScreen.java [hunk 1]
```diff
@@ -208,33 +210,28 @@ private void checkVisible() {
...
-                            @SuppressWarnings("deprecation")
-                            var _unused = imageURL =
-                                    new URL("jar:"+(new File(jarName).toURL().toString())+"!/"+fileName);
-                        } else {
-                            imageURL = new File(fileName).toURL();
-                        }
+                        String jarName = _getImageJarName(splashPtr);
+                        // When calling java -jar, point to the file within the jar
+                        String schema = jarName != null ? "jar:" + jarName + "!/" : "file:";
+                        imageURL = new URI(schema + fileName).toURL();
```
Comment (AlanBateman) [inline src/java.desktop/share/classes/java/awt/SplashScreen.java:230]: This looks broken, I would expect to see File(fileName).toURI() or Path.of(fileName).toUri() to get the URI.

**Comment by TheShermanTanker** [comment]
I'm not sure if assigning an issue to this Pull Request is a good idea when it's potentially broken (not being able to verify that loading images work), tagging @mrserb for guidance

File: src/java.desktop/share/classes/java/awt/SplashScreen.java (hunk 1 above)
Comment (SWinxy) [reply src/java.desktop/share/classes/java/awt/SplashScreen.java:230]: Tried the suggestions, and they are both equivalent (ie `.equals`) to the one in the pull. Would there be something to be concerned about doing it this way?

Comment (AlanBateman) [reply src/java.desktop/share/classes/java/awt/SplashScreen.java:230]: The input for the URI constructor is a URI string, not a file path. Converting from a File (or Path) to a URI requires deal with escaping issues, replacing slashes, etc. You may have to expand your testing to see the issues.

**Pull request closed without merging**